### ISSUE TITLE
SegFault when opening CR2 from Canon Rebel XT

### ISSUE THREAD

**Issue by davidvj** [darktable-org/darktable#16145]
### Describe the bug

The problem occurs when I try to open specific folders.
My data is filed in year sequence .... 2000 to 2224
Each year has one or more sub-folders and the sub-folders in turn contain multiple film-rolls.
I am now finding that a group of older folders, also sub-folders do not load and simply crash the program.
![Screenshot_2024-01-18_09-21-05](https://github.com/darktable-org/darktable/assets/6267270/f6ef0302-4cf3-4bdc-ad12-28bfaf29bd56)
?? Possible relates to images containing 'CR@' Canon files ??

### Steps to reproduce

Go to collections
Pick a folder ... or sub-folder
Double click
Folder opens  without images (trying to load?)
dt then closes.

### Expected behavior

Expected files to appear.

### Logfile | Screenshot | Screencast

I add a dump from -d all
    23.1943 [sql] /var/tmp/pamac-build-david/darktable-git/src/darktable/src/common/image.c:479, function _image_local_copy_full_path(): prepare "SELECT folder || '/' || filename FROM main.images i, main.film_rolls f WHERE i.film_id = f.id AND i.id = ?1"
    23.1948 [exiv2] Directory Canon has an unexpected next pointer; ignored.
/usr/include/c++/13.2.1/optional:477: constexpr _Tp& std::_Optional_base_impl<_Tp, _Dp>::_M_get() [with _Tp = int; _Dp = std::_Optional_base<int, true, true>]: Assertion 'this->_M_is_engaged()' failed.
Aborted (core dumped)

### Commit

4.7.0+314~gff0feabfd2

### Where did you obtain darktable from?

distro packaging

### darktable version

4.7.0+314~gff0feabfd2

### What OS are you using?

Linux

### What is the version of your OS?

Manjaro/Arch rolling release

### Describe your system?

Linux 6.6.10-1
Nvidia -390xx
Mem 16 Gb.
GTK 1.2.10-18

### Are you using OpenCL GPU in darktable?

Yes

### If yes, what is the GPU card and driver?

Nvidia -390xx

### Please provide additional context if applicable. You can attach files too, but might need to rename to .txt or .zip

lua .... image-time

**Comment by davidvj** [comment]
AN UPDATE .... filed 1 hour later.
A collection based on '%CR2' and without any other criteria causes dt to close.

**Comment by gi-man** [comment]
When we had similar issues in the past, there was a corrupted image in one of the folders. You might to add folders individually to find the culprit.

**Comment by davidvj** [comment]
A new test:
1. Copy random .CR2 file to desktop.
2. Test that file opens correctly with Geeqie, gthumb, Viewnior. All open correctly
3. Try to open with dt ..... failure. dt will not start
4. Copy .RAF file to desktop
5. File opens correctly with dt

**Comment by victoryforce** [comment]
> A new test:
> 
> 1. Copy random .CR2 file to desktop.
> 2. Test that file opens correctly with Geeqie, gthumb, Viewnior. All open correctly
> 3. Try to open with dt ..... failure. dt will not start

If this crash is guaranteed to reproduce on some specific file, then please provide this file. All the CR2 files I have load into darktable without problems. So maybe there's something wrong with your files. Without being able to reproduce the crash, there's absolutely nothing we can do to fix it.

You can upload this file to any cloud storage and provide a link. It's even better to zip the CR2 file and attach it here (as long as the file size is less than 25MB if I remember correctly).

**Comment by gi-man** [comment]
And please upload a complete -d common 

What's your path to file ? Nas?

**Comment by davidvj** [comment]
My files are located at /home/david/Pictures.
I used a Canon camera with CR2 format around 2006-7
In the attached test I opened dt normally and then switched to an older folder that contained CR2 files.
dt failed when the switch was made.
[dump1.txt](https://github.com/darktable-org/darktable/files/14002413/dump1.txt)
Typical CR2 file attached also
https://www.dropbox.com/scl/fi/20ie920ua10ur8wia1u1e/IMG_2448.CR2?rlkey=d1arrpw117f4tmubwpknyr3f3&dl=0

**Comment by gi-man** [comment]
My next comment has nothing to do with your issue. I think you should disable opencl in dt for the GPU card in your system. It is using very old drivers (390 vs current 530) and it has 500mb. That amount of memory will force a lot of tiles.

**Comment by davidvj** [comment]
Thank you for the comment on the cpu ..... yes, very old laptop. OK , I will stop the opencl.

**Comment by gi-man** [comment]
I can reproduce your crash with your CR2. I get the same c++ text. I had to rename the file to .old just to get dartktable to not crash. I have plenty of other CR2 files that dont have an issue. This needs more investigation (eg. gdb).

**Comment by davidvj** [comment]
Thank you ..... I have files through 3 years all of which show the same problem. I only found the problem while reworking some material from that prior era.
It is currently not critical for me.

**Comment by gi-man** [comment]
I can get it to open and not crash when using gdb. I busy at the moment. Hopefully someone else can continue.

**Comment by gi-man** [comment]
I think you should edit the title of this issue to: SegFault when opening CR2 from Canon Rebel XT

I can also reproduce using the CR2 from raw.pixls.us. https://raw.pixls.us/getfile.php/1230/nice/Canon%20-%20Digital%20Rebel%20XT%20-%20RAW%20(3:2).CR2 Using my self build in Debug or Release or anything, I cant get it crash. I can only get it to crash if I use the OBS master build. Fedora 39 KDE

```
    18.2010 [avif_open] failed to parse `/home/gman/Pictures/Play Raw/Canon - Digital Rebel XT - RAW.CR2': BMFF parsing failed
    18.2010 [rawspeed_open] extensions list to ignore: `cr3'
/usr/include/c++/13/optional:477: constexpr _Tp& std::_Optional_base_impl<_Tp, _Dp>::_M_get() [with _Tp = int; _Dp = std::_Optional_base<int, true, true>]: Assertion 'this->_M_is_engaged()' failed.
Aborted (core dumped)
```

**Comment by davidvj** [comment]
The problem was discovered recently and I suspect that some 'recent' change within an element of the the git version may have been the cause. Possibly a 2 or 3 month window.
Title changed as requested.

**Comment by davidvj** [comment]
On January 13 I opened a discussion noting for the first time I was seeing a "[Black Border on CR2 images]" (https://discuss.pixls.us/t/black-border-on-cr2-images/41547).
This was probably the last time that I was able to open my CR2 files.
Possibly the fix to that problem has created the current seg fault.
I NOW find a black border (one side) of existing .RAF files.

**Comment by davidvj** [comment]
PROBLEM SOLVED
As suggested I stopped using OpenCl and I am now able to work fully on the CR2 files.

**Comment by gi-man** [comment]
@LebedevRI can you confirm that this issue with CR2 is also fixed with this PR https://github.com/darktable-org/rawspeed/pull/599?

**Comment by LebedevRI** [comment]
Yup.

@TurboGit what's the tentative date for the point release? It would be nice to not miss it for me...

**Comment by LebedevRI** [comment]
Nice, that is what i had in mind too, that is workable.
Thank you.

**Comment by kmilos** [comment]
@davidvj Please note that your camera model [label will change slightly in future versions of dt](https://github.com/darktable-org/rawspeed/pull/618/files) (just in case you were building a large database and collection filters).

**Comment by davidvj** [comment]
4.7.0+361~g7d06f7ce91
Working wiithout (or with)  OpenCl
Unable to access the CR2 files on my system.

**Comment by LebedevRI** [comment]
I'd recommend to use a `Release` build over `RelWithDebInfo` one.

**Comment by LebedevRI** [comment]
17'th is now considered the end of the month? Wow.

**Comment by LebedevRI** [comment]
Right. It's just that somehow i though it would be a week-ish later than 17'th.
17'th may be fine too, but just earlier than what i have though, we'll see.

**Comment by LebedevRI** [comment]
Cool. As i've said, 17'th will probably be fine, but we'll see...

**Comment by LebedevRI** [comment]
(fixed in dt master ad2feeb830e9d3a1b7c00959cdc30339fc7e091a, stable branch is up to the backport maintainer.)
